libcontainer/cgroups/systemd/v1.go

```diff
@@ -216,13 +222,17 @@ func (m *legacyManager) Destroy() error {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
-	dbusConnection, err := getDbusConnection(false)
+	dbusConnection, err := m.dbus.getConnection()
 	if err != nil {
 		return err
 	}
 	unitName := getUnitName(m.cgroups)
 
 	stopErr := stopUnit(dbusConnection, unitName)
+	// reset it if Dbus is closed
```

Comment:
So, `stopUnit` do not ever return a dbus error.

I think it would be better to make it (as well as startUnit) accept `m.dbus` as the first argument, and do the close (and maybe retry) logic within.